hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -142,9 +143,15 @@ public FileIO io() {
   @Override
   protected void doRefresh() {
     String metadataLocation = null;
+    Table table = null;
+
     try {
-      Table table = metaClients.run(client -> client.getTable(database, tableName));
-      HiveOperationsBase.validateTableIsIceberg(table, fullName);
+      table = metaClients.run(client -> client.getTable(database, tableName));
+      HiveOperationsBase.validateTableOrViewIsIceberg(table, fullName);
+
+      if (table.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
```

Comment:
If Hive Table is not iceberg table new metadataLocation should always be null.

Reply:
IIRC the decision was to do not use the HMS tableType for this. Shouldn't we use `BaseMetastoreTableOperations.TABLE_TYPE_PROP` property instead?

Reply:
I have segregated Iceberg [table](https://github.com/apache/iceberg/pull/8907/files#diff-e502621d52f86cf0ec3187dda30ac61f6b76efb7b6276bc8d233ccb2c836fb98R151) and Iceberg [View](https://github.com/apache/iceberg/pull/8907/files#diff-db46657b84d66e084e15f31b8dab21577efb2ae7102863f94c6c9477782de676R83) check here.